internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/NearlineE2ELatencyTest.java

```diff
@@ -0,0 +1,267 @@
+          60,
+          TimeUnit.SECONDS);
+
+      StoreHealthAuditResponse resp = parentControllerCli.listStorePushInfo(storeName, true);
+      LOGGER.info("StoreHealthAuditResponse = {}", resp);
+      Assert.assertFalse(resp.isError());
+      Assert.assertFalse(resp.getRegionPushDetails().isEmpty());
+      for (Map.Entry<String, RegionPushDetails> entry: resp.getRegionPushDetails().entrySet()) {
+        RegionPushDetails detail = entry.getValue();
+        Assert.assertEquals(entry.getKey(), detail.getRegionName());
+        Assert.assertFalse(detail.getPartitionDetails().isEmpty());
+        Assert.assertFalse(detail.getPartitionDetails().get(0).getReplicaDetails().isEmpty());
+      }
+    }
+
+    VeniceClusterWrapper cluster0 = childDatacenters.get(0).getClusters().get(CLUSTER_NAME);
+
+    SystemProducer dc0Producer = getSamzaProducer(cluster0, storeName, Version.PushType.STREAM);
+
+    for (int i = 10; i < 20; i++) {
+      sendStreamingRecord(dc0Producer, storeName, i);
+    }
+    try (AvroGenericStoreClient client = ClientFactory.getAndStartGenericAvroClient(
+        ClientConfig.defaultGenericClientConfig(storeName).setVeniceURL(cluster0.getRandomRouterURL()))) {
+      TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, true, true, () -> {
+        try {
+          for (int i = 10; i < 20; i++) {
+            String key = Integer.toString(i);
+            Object value = client.get(key).get();
+            Assert.assertNotNull(value, "Did not find key " + i + " in store before restarting SN.");
+            Assert.assertEquals(value.toString(), "stream_" + key);
+          }
+        } catch (Exception e) {
+          throw new VeniceException(e);
+        }
+      });
+    }
+
+    AtomicBoolean producerToLocalBroker = new AtomicBoolean(false);
+    AtomicBoolean localBrokerToReadyToServe = new AtomicBoolean(false);
+
+    cluster0.getVeniceServers().forEach(s -> {
+      VeniceServer veniceServer = s.getVeniceServer();
+      MetricsRepository metricsRepository = veniceServer.getMetricsRepository();
+      metricsRepository.metrics().forEach((k, v) -> {
+        if (k.contains(IngestionStats.NEARLINE_PRODUCER_TO_LOCAL_BROKER_LATENCY)) {
+          LOGGER.info("Server: {} , Metric: {}, Value: {}", s.getAddressForLogging(), k, v.value());
+          producerToLocalBroker.set(true);
+        } else if (k.contains(IngestionStats.NEARLINE_LOCAL_BROKER_TO_READY_TO_SERVE_LATENCY)) {
+          LOGGER.info("Server: {} , Metric: {}, Value: {}", s.getAddressForLogging(), k, v.value());
+          localBrokerToReadyToServe.set(true);
+          // Assert.assertTrue(v.value() > 0);
+        }
+      });
+    });
+    Assert.assertTrue(producerToLocalBroker.get());
+    Assert.assertTrue(localBrokerToReadyToServe.get());
+  }
+
+  @AfterClass(alwaysRun = true)
+  public void cleanUp() {
+    Utils.closeQuietlyWithErrorLogged(multiRegionMultiClusterWrapper);
+  }
+
+  private void logMultiCluster() {
```

Comment:
Nice logs! Shall we move it to TestUtils so that others can leverage it too :D

Reply:
Moved to the MutiRegionWrapper class. TestUtils is in different module